sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/v2Commands.scala

```diff
@@ -166,6 +166,52 @@ object OverwritePartitionsDynamic {
   }
 }
 
+case class AppendMicroBatch(
```

Comment:
I don't think we have finalized the streaming writing semantic yet. Ideally, it should be similar to batch (upsert semantic), but we are not at that point yet.

I think we should standardize the streaming writing query plan later. For now, let's just use one query plan with `OutputMode` as the parameter.

Reply:
Oh, I thought only update mode was under discussion. Does overwrite and append have the same issues?

I saw @HeartSaVioR's PR to rename `SupportsStreamingUpdateAsAppend`. Is there a discussion I can take a look at?

Reply:
Streaming write semantic is not same as batch one. The semantic is bound to the stateful operation; there should be only append, update, and truncate and append (complete), and for update we haven't constructed the proper way to define it.

The major concern is that the group keys in stateful operation must be used as keys in update mode. That is currently not possible, but Spark has been dealing with update with the huge risk that we're doing the same as append, and the risk is delegated to the sink (or user). The sink or user has to deal with reflecting the appended output as "upsert". That's why I renamed `SupportsStreamingUpdate` as `SupportsStreamingUpdateAsAppend` to clarify the behavior.

Reply:
Thanks for the context, @HeartSaVioR!

I think we have two ways to proceed:

Option 1: Just adapt `WriteToMicroBatchDataSource ` to use the `Write` abstraction and handle it in `V2Writes`.
Option 2: Define specific plans where we have clarity. For example, `append` and `complete` seem well-defined. We could define plans like `AppendStreamingData` and `TruncateAndAppendStreamingData` or anything like that and have something intermediate for `update`.

I am fine either way but option 1 seems easier for this PR. The rest can be covered by SPARK-27484.

To start with, we should all agree this feature is useful for micro-batch streaming.